server.R

```diff
@@ -719,6 +720,266 @@ shinyServer(function(input, output, session) {
     bayesian_model(sub = TRUE, data(), treatment_list(), input$metaoutcome, input$exclusionbox, 
                    outcome_measure(), input$modelranfix, reference_alter())
   })
+  
+  # Hide the outdated Bayesian visualisations when the data is changed 
+  # or the inputs to the outcome, treatment ranking or model radio buttons are changed
+  observeEvent(c(data(), input$outcomeCont, input$outcomebina, input$rankopts, input$modelranfix), {
```

Comment:
I'm thinking forward to when the app is modularised.
Can you change these blocks such that there are 2 `reactiveVal`s, named something like *model_cleared* and *sub_model_cleared*. These reactives would be set to `TRUE` initially, and when the various inputs change, and would be set to `FALSE` when the button is clicked to run the model.
The `observeEvent` blocks to hide the content would use those reactives as triggers.

When it comes to modularising the code, the 2 reactives wil be passed through to the various modules for each page, and each will be responsible only for hiding its own content.